src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,43 @@
-import React from 'react';
+//TODOS ADD useEffect import, add props in args.
+
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+
+interface Props {
+  selectedTodo?: Todo;
+  setSelectedTodoId: (id: number | null) => void;
+}
+export const TodoModal: React.FC<Props> = props => {
+  const { selectedTodo, setSelectedTodoId } = props;
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    const getUserFromApi = async () => {
+      setIsLoading(true);
+      if (selectedTodo) {
+        try {
+          const userData = await getUser(selectedTodo.userId);
+
+          setUser(userData);
+          setIsLoading(false);
+        } catch (err) {
+          setIsLoading(true);
```

Comment:
Error Handling Issue: The `catch` block incorrectly sets `isLoading` to `true` instead of `false`. This will cause the loader to display indefinitely in case of an error. You should set `isLoading` to `false` here.